Mnt 31 feature make empty block instead of no data

Jade-Dragon88 opened the pull request:
Разработал блок оповещения:
    - при ошибке во время загрузки данных,
    - при удачной загрузке, но с пустыми данными

Jade-Dragon88 committed:
Добавил Empty-блок для отображения ошибки,
если произошёл сбой при загрузке данных.
Добавил Empty-блок для отображения информационного
оповещения при получении пустых данных

Jade-Dragon88 committed:
Разработал блок оповещения:
    - при ошибке во время загрузки данных,
    - при удачной загрузке, но с пустыми данными

ivnvMkhl commented on package.json:
```diff
@@ -15,6 +15,7 @@
     "prepare": "husky install"
   },
   "dependencies": {
+    "@ant-design/icons": "^5.3.6",
```
`"^5.3.6"` -> `"5.3.6"` 

Каретку убираем и фиксируем версию библиотеки чтобы не прилетала другая верися

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -9,25 +9,25 @@ import { notification } from '../../../helpers/notification/notification';
 
 import type { Menti } from '../../../interfaces/menti.interfaces';
+import { Message } from '../../primitives/Message/Message.tsx';
+//import { CloseCircleTwoTone, WarningTwoTone } from '@ant-design/icons';
```
Комменты не оставляем

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -9,25 +9,25 @@ import { notification } from '../../../helpers/notification/notification';
 
   useEffect(() => {
-    fetch('https://imkhl-mentoring-1.glitch.me/menti')
+    fetch('https://imkhl-mentoring-1.glitch.me/menti1111')
```
Урл ломаный оставил - внимательнее!

ivnvMkhl commented on src/components/primitives/Message/Message.tsx:
```diff
@@ -0,0 +1,75 @@
+  //       description={<h1>Данные отсутствуют. Просьба повторить чуть позже</h1>}
+  //     ></AntdEmpty>
+  //   );
+  // }
```
Комменты не оставляем

ivnvMkhl commented on src/components/primitives/Message/Message.tsx:
```diff
@@ -0,0 +1,75 @@
+  };
+
+  const elementDescription = (
+    <span className={styles.description}>{description ? description : DefaultDescriptions[kind]}</span>
```
Вот тут в эту переменную можно положить только тернарник со строкой а span с классом кинуть в разметку

ivnvMkhl commented on src/components/primitives/Message/Message.tsx:
```diff
@@ -0,0 +1,75 @@
+  const iconColor = color ? color : DefaultColors[kind];
+  const iconSize = size ? { fontSize: size } : DefaultSize;
+
+  const getIcon = (kind: Kind) => {
```
Эта функция должна быть вне компоненты тк имеет константное содержимое, но я бы тоже сделал словарик с иконками по kind 
const DefaultIcons = {
   error: <CloseCircle ....>;
   warning: <WarningTwo ....>
}

Jade-Dragon88 committed:
Выделил DefaultIcon
Обьявил  elementIcon

Jade-Dragon88 committed:
Добавил пункты:
    - info для DefaultDescriptions;
    - info для DefaultColors;
    - info для DefaultIcon.

ivnvMkhl commented on src/components/pages/MentiList/MentiList.tsx:
```diff
@@ -9,25 +9,24 @@ import { notification } from '../../../helpers/notification/notification';
 
   useEffect(() => {
-    fetch('https://imkhl-mentoring-1.glitch.me/menti')
+    fetch('https://imkhl-mentoring-1.glitch.me/menti111111')
```
url не исправился

ivnvMkhl commented on src/components/primitives/Message/Message.tsx:
```diff
@@ -0,0 +1,39 @@
+  const { kind, icon, color, description, size } = props;
+  const iconColor = color ? color : DefaultColors[kind];
+  const iconSize = size ? { fontSize: size } : DefaultSize;
+  const DefaultIcon: Record<Kind, JSX.Element> = {
```
Да хорошо, только из компоненты наверх вынеси эту константу чтобы она не создавалась заново при каждом рендере

Jade-Dragon88 committed:
Скорректировал URL menti
